internal/mempool/policy.go

```diff
@@ -111,8 +111,13 @@ func checkInputsStandard(tx *dcrutil.Tx, txType stake.TxType, utxoView *blockcha
 	// but coinbases have already been rejected prior to calling this
 	// function so no need to recheck.
 
+	// Ignore the first input if this is a SSGen (vote) or tspend since
+	// those inputs are not standard by definition.
+	ignoreFirst := txType == stake.TxTypeSSGen ||
+		(isTreasuryEnabled && txType == stake.TxTypeTSpend)
+
 	for i, txIn := range tx.MsgTx().TxIn {
-		if i == 0 && txType == stake.TxTypeSSGen {
+		if i == 0 && ignoreFirst {
```

Comment:
Since `ignoreFirst` is used only once I think you should consider removing it and using the condition directly. Saying this because I don't think the variable name `ignoreFirst` fully describes the condition.

Reply:
I guess it's probably just a matter of taste, but I actually prefer it this way.  Most importantly, it makes it more obvious that the goal is to ignore the first input (at least to me).  However, it also means the checks aren't done multiple times inside the loop.

Reply:
Yeah now that you mention it my suggestion would mean running the check multtiple times in the loop, not worth it. I still think the variable name could be more specific than `ignoreFirst` but i do not feel strongly about it since the descriptive comments above it does compensate for that to some extent.

Reply:
Maybe I could change it to "isVoteOrTSpend". My reasoning for using `ignoreFirst` is that eventually other (future) types of transactions could be ignoring the first txIn as well so "isVoteOrTSpend" wouldn't be strictly true anymore at that point.